pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/TransactionMetadataStore.java

```diff
@@ -138,4 +138,10 @@ default long getLowWaterMark() {
      * @return {@link TxnMeta} the txnMetas of slow transactions
      */
     List<TxnMeta> getSlowTransactions(long timeout);
+
+    /**
+     * set recover end time.
+     * @param time
+     */
+    void setRecoverEndTime(long time);
```

Comment:
Sorry, forgot this one.

I think we don't need to add this one to the interface for now? Only the MLTransactionMetadataStore has the recovery stage. We can just check if it's an MLTransactionMetadataStore, then set the recover end time. Maybe it's not good enough for now, we can try to refactor it in the future.

Reply:
you are right. i have fix it.